### ISSUE TITLE
checkcommits: use TRAVIS_PULL_REQUEST_SHA

### ISSUE THREAD

**Issue by bergwolf** [kata-containers/tests#177]
We've got quite a few checkcommits failures like the following:
```
Detected TravisCI Environment
ERROR: Failed to run command /usr/bin/git rev-list --no-merges --reverse master..8e1f6d1b64742dfad80b6132fb4daad0ea4282ee: exit status 128 (stdout: , stderr: fatal: Invalid revision range master..8e1f6d1b64742dfad80b6132fb4daad0ea4282ee
)
ERROR: checkcommits failed. See the document below for help on formatting
```
Add some debug log to properly identify the cause.

**Comment by grahamwhaley** [comment]
Hi @bergwolf - I think @jodh-intel can give you the full details and the pointer to the open Travis bug report for this - we have tried to track down and fix this before for CC, and could not fix it :-(  Hence, we are rapidly trying to move everything over to Jenkins (which is what we did for CC).

**Comment by bergwolf** [comment]
@grahamwhaley my guess is that the commit `8e1f6d1b64742dfad80b6132fb4daad0ea4282ee` came from nowhere. The same command fails in my local environment as well. Therefore I created the PR to confirm it. If we are sure that it is the case, maybe we should stop relying on `TRAVIS_COMMIT` to get the testing commit.

**Comment by jodh-intel** [comment]
Hi @bergwolf - yes, the problem we believe is https://github.com/travis-ci/travis-ci/issues/7830.

I'd very much like to switch off Travis and use Jenkins entirely due to that bug (and the lack of VT-x). See https://github.com/kata-containers/tests/issues/132 for the last "blocker" to move away from Travis).

In summary, printing env vars won't help debug this particular problem. However, I do think there is a lot we can to do make debugging CI failures clearer generally, including dumping the full environment using my `procenv` tool and using the Jenkins pipeline plugin:

- https://github.com/clearcontainers/jenkins/issues/5
- https://github.com/clearcontainers/jenkins/issues/7

**Comment by jodh-intel** [comment]
Those last 2 issue re-raised in Kata as:

- https://github.com/kata-containers/ci/issues/14
- https://github.com/kata-containers/ci/issues/15

**Comment by bergwolf** [comment]
@jodh-intel Thanks for the links. We should be able to get the testing commit head on our own rather than relying on travis environment variables. Then the issue would go away.

**Comment by jodh-intel** [comment]
@bergwolf - the "fix" for Travis is to "force rebase" (change `pick` to `reword`), then re-force push. If that doesn't work, you have to close the PR and open a new one from our experiences.

**Comment by jodh-intel** [comment]
@bergwolf - The git environment in Travis is a bit "weird" - there is only a single branch when testing a PR... and that *PR* branch is called... `master`! So, the fix you're referring to would be to add a remote and a `git fetch` which I didn't want to do for `checkcommits`. Some more details here:

- https://github.com/kata-containers/tests/blob/master/cmd/checkcommits/checkcommits.go#L433..L478

Bottom line: we shouldn't use Travis as it's too limited for our requirements imho.

**Comment by jodh-intel** [comment]
If you want to see what variables Travis provides for a PR, take a look at:

- https://travis-ci.org/jamesodhunt/procenv/jobs/233252731

and search for "environment:".

**Comment by bergwolf** [comment]
@jodh-intel travis actually merges the testing branch with master to make sure there is no conflict. That's why there is only one master branch.

```
$ git clone --depth=50 https://github.com/kata-containers/tests.git kata-containers/tests
Cloning into 'kata-containers/tests'...
remote: Counting objects: 1557, done.
remote: Compressing objects: 100% (919/919), done.
remote: Total 1557 (delta 542), reused 1551 (delta 539), pack-reused 0
Receiving objects: 100% (1557/1557), 1.49 MiB | 22.51 MiB/s, done.
Resolving deltas: 100% (542/542), done.
$ cd kata-containers/tests
0.23s$ git fetch origin +refs/pull/173/merge:
remote: Counting objects: 6, done.
remote: Compressing objects: 100% (2/2), done.
remote: Total 6 (delta 4), reused 5 (delta 4), pack-reused 0
Unpacking objects: 100% (6/6), done.
From https://github.com/kata-containers/tests
 * branch            refs/pull/173/merge -> FETCH_HEAD
```

Looking at [Travis-ci Default Environment Variables](https://docs.travis-ci.com/user/environment-variables/#Default-Environment-Variables), it seems that we can also use `TRAVIS_PULL_REQUEST_SHA` instead of `TRAVIS_COMMIT` in checkcommits. Have you tried that with CC before?

```
TRAVIS_PULL_REQUEST_SHA:
if the current job is a pull request, the commit SHA of the HEAD commit of the PR.
if the current job is a push build, this variable is empty ("").
```

**Comment by jodh-intel** [comment]
I don't think I did. However, I don't personally plan to change checkcommits since we moved away from Travis in Clear Containers and I don't see the value in staying with it for Kata either due to the aforementioned limitations.

**Comment by bergwolf** [comment]
Fair enough. I can try it. Just a one liner ;)
